workflow.md

```diff
@@ -100,10 +123,91 @@ Please note that actual version of project is defined by project settings,
 not by release branch name. So we must keep this correspondence. For example,
 there's `cardano-sl.cabal` file for Cardano SL project, and this file contains
 `version` parameter. So when we release `0.2.0` version, we must change value
-of this parameter to `0.2.0`.
+of this parameter to `0.2.0`. If project consists of multiple
+packages, all of them should have the same version.
 
 Please keep the value of `version` parameter in the `master` branch actual
-as well.
+as well (corresponding to the latest release).
+
+### Special workflow for cardano-sl project
+
+Table below illustrates a special workflow used for cardano-sl project
+with respect to branches and deployment clusters.
+
+| Branch             | Deployment  | Purpose                                        |
+| ------------------ | ----------- | ---------------------------------------------- |
+| master             | csl-testing | Testing current master (by devs)               |
+| cardano-sl-X.Y-rc  | csl-qa      | QA testing of public release candidate (qanet) |
+| cardano-sl-X.Y     | csl-testnet | Public release (testnet)                       |
+| cardano-sl-X.Y-mid | ——————————— | Used by other projects (e. g. explorer)        |
+
+Flow should be following:
+1. When version is being wrapped up, `master` should be deployed to
+   `csl-testing` cluster, problems arising right away should be
+   resolved.
+2. After successful deployment on `csl-testing`:
+    a) `vX.Y.0` tag should be set (in `master`).
+    b) `cardano-sl-X.Y-rc` should be branched out of `master`.
+    c) `cardano-sl-X.Y-rc` should be deployed on `csl-qa` cluster.
+    d) All bugs discovered should be fixed in `-rc` branch, version
+	is increased for each fix.
+3. After `X.Y` version is polished and is ready to be published (it's
+`vX.Y.Z` already):
+    a) `cardano-sl-X.Y-rc` should be renamed to `cardano-sl-X.Y`, `cardano-sl-X.Y-mid` immediately spawned
+    b) `cardano-sl-X.Y-mid` is immediately spawned from the same commit.
+    c) `genesis.bin`, `constants-*.yaml` should be changed.
+    d) `vX.Y.Z` should be deployed on `csl-testnet` cluster.
+    e) tada.iohk.io, daedaluswallet.io should be updated.
+
+Rules for `-rc` branch are same as for release branch, specifically:
+* `-rc` branch must be merged into `master` after every change.
+* `master` is never merged into `-rc` branch.
+* `-rc` branch must be protected.
+
+### Interaction between cardano-sl and projects dependent on it
+
+Some projects (e. g. `cardano-sl-explorer`, `daedalus`) depend on
+`cardano-sl`. In order to facilitate interaction with `cardano-sl` we
+use special branch `-mid` mentioned above. Explorer is used as an
```

Comment:
Yes, that was my initial understanding and I still think so. Maybe @georgeee had a different vision when he proposed it.